reo/src/reopt.jl

```diff
@@ -10,8 +10,7 @@ function add_continuous_variables(m, p)
     	dvSystemSizeSegment[p.Tech, p.Subdivision, p.Seg] >= 0   #X^{\sigma s}_{tks}: System size of technology t allocated to segmentation k, segment s [kW]  (NEW)
 		dvGridPurchase[p.PricingTier, p.TimeStep] >= 0   # X^{g}_{uh}: Power from grid dispatched to meet electrical load in demand tier u during time step h [kW]  (NEW)
 	    dvRatedProduction[p.Tech, p.TimeStep] >= 0   #X^{rp}_{th}: Rated production of technology t during time step h [kW]  (NEW)
-	    dvProductionToGrid[p.Tech, p.SalesTiers, p.TimeStep] >= 0  # X^{ptg}_{tuh}: Exports from electrical production to the grid by technology t in demand tier u during time step h [kW]   (NEW)
-	    dvStorageToGrid[p.StorageSalesTiers, p.TimeStep] >= 0  # X^{stg}_{uh}: Exports from electrical storage to the grid in demand tier u during time step h [kW]  (NEW)
+		dvProductionToCurtail[p.Tech, p.TimeStep] >= 0
```

Comment:
Should we do a similar conditional where this decision variable is only created in the case that some Tech "can_curtail". I.e. if no techs can curtail, avoid creating this dv.

Reply:
Stringing along these comments as I review, sorry... I now see there is p.TechsCannotCurtail - could we check if this is empty to avoid creating dvProductionToCurtail or would indexing dvProductionToCurtail[{p.Techs} - {p.TechsCannotCurtail}, p.TimeStep] make the dv "Null" if none of the p.Techs can curtail?  Otherwise, I'm fine to keep your "force to zero" constraint for this dV if Tech is in p.TechsCannotCurtail.

Reply:
Yeah, I think the custom types (e.g. ExportAndCurtailTechs) is a good idea instead of more if/else. However, I also see the advantage of not adding redundant code to capture all the combinations (at the expense of possibly very small amounts of solve time). I don't have a strong opinion on keeping the current modifications versus adding the new Tech types.

Reply:
I now understand better your reference to Julia's method dispatch (I read about it in Julia's documentation after responding). It seems like a pretty smart strategy to start incorporating it, and Julia claims it is one of its most powerful/unique features. I think it would be valuable to introduce that to the model for this use case, and others could learn and re-use that strategy as we develop the model and think about speed-up options. So, without knowing how much, if any, additional and duplicative code this would add, I think it would be beneficial to implement. I'm open to discuss further if need be.

Reply:
@Bill-Becker I just [pushed an outline of the changes needed to use the method dispatch for defining constraints](https://github.com/NREL/REopt_Lite_API/pull/156/commits/d4f0b5a35c7bd603a216a77297da3796194b2b3a). It does not create redundant code (the `...` above are the same code in both the if/else structure or the method dispatch structure).  Once I developed the outline I realized that it will create significant changes to utils.jl and reopt.jl, and so I recommend that we hold off until chpbeta branch is merged into develop branch.

Reply:
@NLaws Sounds good. Thanks for creating the outline. I don't think it's even "must have" before the public release (early/mid January code freeze) since there's a bunch of other stuff we need to make sure we get in there and tested before then.